packages/webapp/webapp_server.js

```diff
@@ -345,8 +345,33 @@ function getBoilerplate(request, arch) {
   return getBoilerplateAsync(request, arch).await();
 }
 
+WebApp.encodeRuntimeConfig = function (rtimeConfig) {
+  return JSON.stringify(encodeURIComponent(JSON.stringify(rtimeConfig)));
+}
+
+WebApp.decodeRuntimeConfig = function (rtimeConfig) {
+  return JSON.parse(decodeURIComponent(JSON.parse(rtimeConfig)));
+}
+
+const runtimeConfig = {
+  'hooks': [],
+  'update': {}
+};
+
+WebApp.addRuntimeConfigHook = function (hook) {
+  if(typeof hook !== 'function') throw new Error('WebApp.addRuntimeConfigHook must be a function');
+  runtimeConfig.hooks.push(hook);
+}
+
 function getBoilerplateAsync(request, arch) {
-  const boilerplate = boilerplateByArch[arch];
+  let boilerplate = boilerplateByArch[arch];
+  _.each(runtimeConfig.hooks, function (hook) {
+    const meteorRuntimeConfig = hook(arch, request, boilerplate.baseData.meteorRuntimeConfig, runtimeConfig.update[arch]);
```

Comment:
I would suggest a single object param here so next changes would be easier without breaking changes due to the order.

```suggestion
    const meteorRuntimeConfig = hook({ arch, request, encodedCurrentConfig: boilerplate.baseData.meteorRuntimeConfig, updated: runtimeConfig.update[arch] });
```

Reply:
Good feedback, thanks! done.